contentcuration/contentcuration/static/js/administration/router.js

```diff
@@ -0,0 +1,293 @@
+		return false
+	},
+	getRoute({name, filter, sortKey, sortOrder, search, page, pageSize}){
+		return name +
+				(filter ? "/filter/" + filter : "") +
+				(	
+					(sortKey || sortOrder) ? 
+					"/sort/" + (sortKey ? sortKey : GET_DEFAULT_ROUTES()[name]['sortKey']) + "-" +
+					(sortOrder ? sortOrder : "ascending")
+					: ""
+				) +
+				(search ? "/search/" + search : "") +
+				(page ? "/p" + page : "") +
+				(pageSize ? "/" + pageSize + "-per-page" : "")
+	},
+	gotoRouteForParams(params){
+		if (!params.name) {
+			// update existing params with incoming ones
+			let currentRouteParams = Object.assign({}, this.currentRouteParams)
+			params = Object.assign(currentRouteParams, params)
+		}
+		let route = this.getRoute(params)
+		this.navigate(route, {trigger: true})
+	},
+	gotoPage(page){
+		this.gotoRouteForParams({page: page})
+	},
+	gotoNextPage(){
+		this.gotoPage(this.currentRouteParams.page+1)
+	},
+	gotoPreviousPage(){
+		this.gotoPage(this.currentRouteParams.page-1)
+	},
+	initialize: function(){
+		var AdministrationView = require("./views");
+		this.admin_view = new AdministrationView.AdminView ({
+			el: $("#admin-container"),
+			router: this,
+		});
+		this.collections = {}
+		this.currentRouteParams = GET_DEFAULT_ROUTES()['channels']
+		this.routeParamsCache = GET_DEFAULT_ROUTES()
+		let router = this;
+		$('.nav-tabs a').click(function(e){
+			e.stopImmediatePropagation()
+			let routeParams = router.routeParamsCache[e.currentTarget.attributes['data-href'].value]
+			router.gotoRouteForParams(routeParams)
+		})
+		window.current_user = new Models.UserModel(window.user);
+	},
+	getSelected(opts) {
+		for (let k in opts){
+			if (opts[k].selected === true) {
+				return k
+			}
+		}
+	},
+	updateCollectionStateFromParams(collection, filter, sortKey, order, search, page = 1, pageSize){
+		collection.state.currentPage = page ? Number(page) : 1
+		collection.state.pageSize = pageSize ? Number(pageSize) : collection.state.pageSize
+
+		if (filter) {
+			for (let k in collection.filterOptions){
+				collection.filterOptions[k].selected = k === filter
+			}
+			collection.state.filterQuery = collection.filterOptions[filter].queryParams
+		}
+
+		// if (search) {
+		collection.state.search = search
+		collection.state.filterQuery.search = search
+		// }
+
+		if (order) {
+			for (let k in collection.sortOrderOptions){
+				collection.sortOrderOptions[k].selected = k === order
+			}
+		}
+		
+
+		if (sortKey) {
+			for (let k in collection.sortFilterOptions){
+				collection.sortFilterOptions[k].selected = k === sortKey
+			}
+		}
+
+		if (sortKey || order) {
+			let oldSortKey
+			try {
+				oldSortKey = collection.state.sortKey
+			} catch (error) {
+				oldSortKey = getSelection(collection.sortFilterOptions)
+			}
+										
+			collection.state.order = order ? (order === "ascending" ? -1 : 1) : collection.state.order
+			collection.state.sortKey = sortKey ? sortKey : oldSortKey
+		}
+	},
+	default: function(){
```

Comment:
The default route handler just waits till the next frame and redirects to "/channels/"